File: Data Structures & Algorithms/max-consecutive-ones/submission-1.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int count = 0;
        int max_count = 0;
        for (int i = 0; i<nums.length; i++) {
            System.out.println("i : "+i+", count: "+count+", max_count: "+max_count);
            if (nums[i]==0) {
                max_count = Math.max(count, max_count);
                count = 0;
            }
            else {
                count++;
            }
        }

        return Math.max(count, max_count);
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0, r = 0, maxLen = 0, maxFreq = 0;
        unordered_map<char, int> count;

        while (r < s.length()) {
            count[s[r]]++;
            maxFreq = max(maxFreq, count[s[r]]);

            while ((r-l+1) - maxFreq > k) count[s[l++]]--;
            maxLen = max(maxLen, r - l + 1);
            r++;
        }
        return maxLen;
    }
};
